.github/build-images.java

```diff
@@ -0,0 +1,224 @@
+//DEPS com.github.docker-java:docker-java-transport-okhttp:3.2.5
+//DEPS com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.11.2
+//DEPS org.slf4j:slf4j-api:1.7.30
+//DEPS org.slf4j:slf4j-simple:1.7.30
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.model.Image;
+import com.github.dockerjava.api.model.PruneType;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientConfig;
+import com.github.dockerjava.core.DockerClientImpl;
+import com.github.dockerjava.okhttp.OkDockerHttpClient;
+import com.github.dockerjava.transport.DockerHttpClient;
+import org.zeroturnaround.exec.ProcessExecutor;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.Callable;
+
+@Command(name = "build-images", mixinStandardHelpOptions = true,
+        description = "build container images providing the `native-image` executable for Quarkus")
+class BuildImages implements Callable<Integer> {
+
+    @CommandLine.Parameters(index = "0", description = "A yaml file containing the configuration of the images to build.")
+    private File config;
+
+    public static void main(String... args) {
+        int exitCode = new CommandLine(new BuildImages()).execute(args);
+        System.exit(exitCode);
+    }
+
+    @Override
+    public Integer call() throws IOException {
+
+        if (!config.isFile()) {
+            System.out.println("The configuration file " + config.getAbsolutePath() + " does not exist - exiting");
+            return -1;
+        }
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        mapper.findAndRegisterModules();
+        Configuration configuration = mapper.readValue(config, Configuration.class);
+        validateConfiguration(configuration);
+
+        // Preparation
+        Docker docker = new Docker();
+
+        // Build
+        for (String version : configuration.versions) {
+            docker.deleteExistingImageIfExists(configuration.imageName, version);
+
+            try {
+                build(version, configuration);
+            } catch (Exception e) {
+                String name = configuration.imageName + ":" + version;
+                System.out.println("Build of image  " + name + " has failed: " + e.getMessage() + " - exiting");
+                return -1;
+            }
+        }
+
+        // Post-Validation
+        if (!docker.validateCreatedImages(configuration)) {
+            return -1;
+        }
+
+        if (!docker.createTags(configuration)) {
+            return -1;
+        }
+
+        // Cleanup
+        docker.prune();
+
+        return 0;
+    }
+
+    private void validateConfiguration(Configuration configuration) {
```

Comment:
Hmm, this should be moved in Configuration. I 'll do it after the tests complete :)